resolver-dns/src/main/java/io/netty/resolver/dns/BiDnsQueryLifecycleObserver.java

```diff
@@ -0,0 +1,96 @@
+package io.netty.resolver.dns;
+
+import io.netty.channel.ChannelFuture;
+import io.netty.handler.codec.dns.DnsQuestion;
+import io.netty.handler.codec.dns.DnsResponseCode;
+import io.netty.util.internal.UnstableApi;
+
+import java.net.InetSocketAddress;
+import java.util.List;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+
+/**
+ * Combines two {@link DnsQueryLifecycleObserver} into a single {@link DnsQueryLifecycleObserver}.
+ */
+@UnstableApi
+public final class BiDnsQueryLifecycleObserver implements DnsQueryLifecycleObserver {
```

Comment:
Just an optimization to avoid allocation of collections/arrays. My rational was we can always add a `MultiDnsQueryLifecycleObserver` later ... does this sound reasonable?